tests/suit/framework/ovirt.go

```diff
@@ -0,0 +1,257 @@
+func (r *OvirtClient) getNicsFromVM(vmRef ref.Ref) (nicIds []string, err error) {
+	_, vmService, err := r.getVM(vmRef)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get VM - %v", err)
+	}
+
+	nicsResponse, err := vmService.NicsService().List().Send()
+	nics, ok := nicsResponse.Nics()
+	if !ok {
+		return nil, fmt.Errorf("Failed to get nics")
+	}
+
+	for _, nic := range nics.Slice() {
+		vnicService := r.connection.SystemService().VnicProfilesService().ProfileService(nic.MustVnicProfile().MustId())
+		vnicResponse, err := vnicService.Get().Send()
+		if err != nil {
+			return nil, fmt.Errorf("Failed to get vnic service = %v", err)
+		}
+		profile, ok := vnicResponse.Profile()
+		if !ok {
+			return nil, fmt.Errorf("Failed to get nic profile")
+		}
+		network, ok := profile.Network()
+		if !ok {
+			return nil, fmt.Errorf("Failed to get network")
+		}
+		networkId, ok := network.Id()
+		if !ok {
+			return nil, fmt.Errorf("Failed to get network id")
+		}
+		nicIds = append(nicIds, networkId)
+	}
+	return
+}
+
+// getSDFromVM - get storage domains from specific VM
+func (r *OvirtClient) getSDFromVM(vmRef ref.Ref) (storageDomains []string, err error) {
+	_, vmService, err := r.getVM(vmRef)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get VM - %v", err)
+	}
+
+	diskAttachementResponse, err := vmService.DiskAttachmentsService().List().Send()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get disk attachment service  %v", err)
+	}
+	disks, ok := diskAttachementResponse.Attachments()
+	if !ok {
+		return nil, fmt.Errorf("Failed to get disks")
+	}
+	for _, da := range disks.Slice() {
+		disk, ok := da.Disk()
+		if !ok {
+			return nil, fmt.Errorf("Failed to get disks")
+		}
+		diskService := r.connection.SystemService().DisksService().DiskService(disk.MustId())
+		diskResponse, err := diskService.Get().Send()
+		if err != nil {
+			return nil, fmt.Errorf("Failed to get disks - %v", err)
+		}
+		sds, ok := diskResponse.MustDisk().StorageDomains()
+		if !ok {
+			return nil, fmt.Errorf("Failed to get storage domains")
+		}
+		for _, sd := range sds.Slice() {
+			sdId, ok := sd.Id()
+			if !ok {
+				return nil, fmt.Errorf("Failed to get storage domain id")
+			}
+			storageDomains = append(storageDomains, sdId)
+		}
+	}
+	return
+}
+
+func (r *OvirtClient) GetVMNics() []string {
+	return r.vmData.nicPairs
+}
+
+func (r *OvirtClient) GetVMSDs() []string {
+	return r.vmData.sdPairs
+}
+
+func (r *OvirtClient) GetTestVMId() string {
+	return r.testVMId
+}
+
+// Close the connection to the oVirt API.
+func (r *OvirtClient) Close() {
+	if r.connection != nil {
+		_ = r.connection.Close()
+		r.connection = nil
+	}
+}
+
+// OvirtClient - oVirt VM Client
+type OvirtClient struct {
+	connection   *ovirtsdk.Connection
+	Cacert       []byte
+	Username     string
+	OvirtURL     string
+	Password     string
+	testVMId     string
+	storageClass string
+	vmData       OvirtVM
+	CustomEnv    bool
+}
+
+type OvirtVM struct {
```

Comment:
i would put everything related to the vm within the VM struct, including the VM ID from above (the ovirt client).

Reply:
yeap make sense

Reply:
done see https://github.com/kubev2v/forklift/pull/129/commits/2b0b4139f68e2f8c60ccb636993d8c9a9fc5f440